src/atomate2/openmm/jobs/base.py

```diff
@@ -0,0 +1,528 @@
+    makers. It includes common attributes and methods for setting up,
+    running, and closing OpenMM simulations. Subclasses can override
+    the run_openmm method to define specific simulation logic.
+
+    In general, any missing values will be taken from the
+    previous task, if possible, and the default values defined in
+    atomate2.openmm.OPENMM_MAKER_DEFAULTS, if not.
+
+    Attributes
+    ----------
+    name : str
+        The name of the OpenMM job.
+    tags : Optional[List[str]]
+        Tags for the OpenMM job.
+    n_steps : Optional[int]
+        The number of simulation steps to run.
+    step_size : Optional[float]
+        The size of each simulation step (picoseconds).
+    temperature : Optional[float]
+        The simulation temperature (kelvin).
+    friction_coefficient : Optional[float]
+        The friction coefficient for the
+        integrator (inverse picoseconds).
+    platform_name : Optional[str]
+        The name of the OpenMM platform to use, passed to
+        Interchange.to_openmm_simulation.
+    platform_properties : Optional[dict]
+        Properties for the OpenMM platform,
+        passed to Interchange.to_openmm_simulation.
+    state_interval : Optional[int]
+        The interval for saving simulation state.
+        To record no state, set to 0.
+    state_file_name : Optional[str]
+        The name of the state file to save.
+    traj_interval : Optional[int]
+        The interval for saving trajectory frames. To record
+        no trajectory, set to 0.
+    wrap_traj : Optional[bool]
+        Whether to wrap trajectory coordinates.
+    report_velocities : Optional[bool]
+        Whether to report velocities in the trajectory file.
+    traj_file_name : Optional[str]
+        The name of the trajectory file to save.
+    traj_file_type : Optional[str]
+        The type of trajectory file to save. Supports any output format
+        supported by MDAnalysis.
+    """
+
+    name: str = "base openmm job"
+    tags: list[str] | None = field(default=None)
+    n_steps: int | None = field(default=None)
+    step_size: float | None = field(default=None)
+    temperature: float | None = field(default=None)
+    friction_coefficient: float | None = field(default=None)
+    platform_name: str | None = field(default=None)
+    platform_properties: dict | None = field(default=None)
+    state_interval: int | None = field(default=None)
+    state_file_name: str | None = field(default=None)
+    traj_interval: int | None = field(default=None)
+    wrap_traj: bool | None = field(default=None)
+    report_velocities: bool | None = field(default=None)
+    traj_file_name: str | None = field(default=None)
+    traj_file_type: str | None = field(default=None)
+    embed_traj: bool | None = field(default=None)
+
+    @openff_job
+    def make(
+        self,
+        interchange: Interchange | OpenMMInterchange | bytes,
+        prev_task: OpenMMTaskDocument | None = None,
```

Comment:
Yeah can do. I somewhat dislike having `prev_dir` because you have no idea what needs to be in the directory. But for the sake of minimizing database calls and consistency with other codes, I do think it makes sense.